modules-js/react-fleet/src/components/Header.tsx

```diff
@@ -6,10 +6,47 @@ export default function Header(
   props: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
 ) {
   let className = 'h';
+  let isScrolling;
+
   if (props.className) {
     className = `${className} ${props.className}`;
   }
 
+  // When the City seal is visible in the header, it should slide halfway out
+  // of view when a user scrolls past ~310px. When scrolling back towards the
+  // top, the seal should slide back to its original position.
+  function handleScroll(): void {
+    window.clearTimeout(isScrolling);
+
+    isScrolling = window.setTimeout(() => handleScrollStop(), 1000);
+  }
+
+  function handleScrollStop(): void {
+    const sealElement = document.getElementById('seal');
+
+    if (sealElement && isSealVisible()) {
+      if (window.pageYOffset > 310) {
+        sealElement.classList.add('s--u');
+      } else {
+        sealElement.classList.remove('s--u');
+      }
+    } else {
+      // The listener is unnecessary if the seal is not visible, so remove it.
+      window.removeEventListener('scroll', handleScroll);
+    }
+  }
+
+  window.addEventListener('scroll', handleScroll);
```

Comment:
This is likely adding a bunch of event listeners over the course of the app.

I think we need to do this in a hook or change to a class component and do it with the lifecycle events.